### ISSUE TITLE
Not working in 2.2?

### ISSUE THREAD

**Issue by FryDay** [cocodelabs/znc-palaver#28]
This module seems to be having issues with Palaver 2.2. The client never seems to receive push notifications. My same ZNC installation was working fine pre 2.2.

**Comment by kylef** [comment]
@FryDay Can you include a bit of information:

- Version of ZNC
- Version of the module
- Is the client registered with ZNC (/msg *palaver info) from the client

**Comment by FryDay** [comment]
ZNC 1.6.1

Latest version of the module from git. (I am also using clientaway and verified that my nick is away before testing).

Info: Palaver ZNC: d0a8db1
Current device: (long string of characters)
2 registered devices (my wife and I both use it).

Let me know if you need anything else.

**Comment by kylef** [comment]
@FryDay Does the following send you a push notification.

From a different client while Palaver is not connected and in the background:

```
/msg *palaver test
```

**Comment by FryDay** [comment]
It does not. It says "Notification sent to 1 clients." but nothing is received.

**Comment by emagaliff** [comment]
@kylef Same issue as @FryDay 

Was not working before 2.2  update either

**Comment by kylef** [comment]
@emagaliff Can you make sure that you have the latest version of the module and show the output of `/msg *palaver info`.

@FryDay Can you try going into "iOS setting app -> Palaver > Notifications" and toggle "Allow Notifications off and on and then go back into Palaver. See if this solves your problem.

**Comment by FryDay** [comment]
@kylef Didn't work.

**Comment by FryDay** [comment]
@kylef Found something that might be relevent. On the wiki, it says:
Notifications will only be sent if all your clients are disconnected, or marked as away.

If I run a list right now, I see this:

*palaver> +==========================================+===============+==========+=============+
<*palaver> | Device                                   | User          | Network  | Negotiating |
<*palaver> +==========================================+===============+==========+=============+
<*palaver> | aebcb9d98b9c3acdaa83e1bcef170cc6984ffb70 | jordan        | freenode | false       |
<*palaver> +------------------------------------------+---------------+----------+-------------+
<*palaver> | aebcb9d98b9c3acdaa83e1bcef170cc6984ffb70 | jordan_phone  | freenode | false       |
<*palaver> +------------------------------------------+---------------+----------+-------------+
<*palaver> | 349b59f5d8b99e5677a5ea58d5c51fa36264fd17 | jessica_phone | freenode | false       |
<*palaver> +------------------------------------------+---------------+----------+-------------+
<*palaver> | e649d13fe0944c37fc11e3bcb1dec0593c581570 | jordan_phone  | freenode | false       |
<*palaver> +------------------------------------------+---------------+----------+-------------+
<*palaver> | dc4349fc1be78866350637e882cc9166a713811a | jordan_phone  | freenode | false       |
<*palaver> +------------------------------------------+---------------+----------+-------------+
<*palaver> | 3b59a40541f63e1f4a2fb612d78c6829751cdf8b | jessica_phone | freenode | false       |
<*palaver> +==========================================+===============+==========+=============+

When do these get cleaned up? Could there be a conflict of some sort?

**Comment by aarondunlap** [comment]
@FryDay the test should still send a push notification to palaver even if you're not marked as away.  Just tested this on my iPhone 6 with znc 1.7.x

**Comment by FryDay** [comment]
@aarondunlap And your push notifications are working as well? I feel like I shouldn't have two separate users with identical device numbers. It doesn't appear like there is a way to unregister a device, though.

**Comment by emagaliff** [comment]
@kylef Just rebuilt the module from the latest github source. Output:

`[11:14:13]  <*palaver>	Palaver ZNC: unknown -- http://palaverapp.com/
[11:14:13]  <*palaver>	1 registered devices
[11:14:13]  <*status>	ZNC 1.6.1 - http://znc.in
[11:14:13]  <*status>	IPv6: yes, SSL: yes, DNS: threads, charset: yes`

**Comment by kylef** [comment]
Hi everyone. This thread is a bit hard to follow with three people latching on with separate problems.

Can anyone who is still having problems move to email (support@palaverapp.com) or hit me up on Freenode (kylef).

### Steps to confirm

- [ ] You are running the latest version of Palaver module (Palaver ZNC: d0a8db1 or 1.0.0). Note if you upgraded by unloading the old module and loading the new one. The old one may still be in memory and the only way to confirm it worked is by checking the version in the module via `/msg *palaver info`.
- [ ] You are running ZNC 1.6.0 or newer.
- [ ] You device is registered with ZNC (`/msg *palaver info` from the device)
- [ ] If your device is registered, using `/msg *palaver test` deliveries a notification to your registered device

**NOTE**: *Do NOT share tokens from `/msg *palaver info` and `/msg *palaver list` anywhere but privately with support@palaverapp.com. These are private access tokens and should not be shared with anyone.*

#### Debugging Steps

###### My Device is not registered when I use `/msg *palaver info` from the device.

Please go into Palaver's settings and then tap "System Settings" followed by "Notifications" and toggled "Allow Notifications" off and on and then go back into the app and try `/msg *palaver info` again.


###### My Device is registered and `/msg *palaver test` is not showing the notification.

This is a tricky problem and can be due to a bug in older versions of OpenSSL. The only way to dig into this one further is to run ZNC in debug mode (`znc --debug`) and look at the output produced when you run `/msg *palaver test`. The output should include the exact reason for the push notification failure. You can send me the output for analysis.

---

Note the output of `/msg *palaver list` is for debugging purposes, and old stale devices are not a problem and will be cleaned up on their own in the future.

**Comment by kylef** [comment]
I've also added some other reasons for failings inside the [README](https://github.com/cocodelabs/znc-palaver#common-reasons).
